### PR TITLE
Fix kernel to setup parameters again and refactor isArray

### PR DESCRIPTION
**Pull request by abhisheksoni27**

### CODE REVIEW

File: src/core/utils.js [hunk 1]
```diff
@@ -243,21 +243,11 @@ class Utils extends UtilsCore {
...
-			case Uint16Array:
-			case Int32Array:
-			case Uint32Array:
-			case Float32Array:
-			case Float64Array:
-			case Array:
-				return true;
-			default:
-				return false;
+		if (array.length) {
```
Comment (robertleeplummerjr) [inline src/core/utils.js]: I don't think this works on two accounts:
`utils.isArray('hi mom') -> true`
`utils.isArray([]) -> false`

Comment (abhisheksoni27) [inline src/core/utils.js]: Should we worry about the first one? :laughing: 

Fixed.

File: src/core/utils.js [hunk 2]
```diff
@@ -243,21 +243,11 @@ class Utils extends UtilsCore {
...
-			case Uint16Array:
-			case Int32Array:
-			case Uint32Array:
-			case Float32Array:
-			case Float64Array:
-			case Array:
-				return true;
-			default:
-				return false;
+		if((array.length || array.length === 0) && (array.constructor !== String)) {
```
Comment (robertleeplummerjr) [inline src/core/utils.js]: How about: `if (array.hasOwnProperty('length'))`

Comment (abhisheksoni27) [inline src/core/utils.js]: Good catch. But it, unfortunately, fails with `Strings`.

Comment (robertleeplummerjr) [inline src/core/utils.js]: Really?  Typed arrays?

Comment (robertleeplummerjr) [inline src/core/utils.js]: wow, mind blown

Comment (robertleeplummerjr) [inline src/core/utils.js]: What do you think about keeping this existing functionality, and adding a method that allows them to add their array type, that could iterate through, and if it exists we could return true.  That way it is all strongly typed.

Comment (abhisheksoni27) [inline src/core/utils.js]: Indeed. 
screenshot from 2017-07-15 23-45-03

Comment (robertleeplummerjr) [inline src/core/utils.js]: I ran it locally, and my mind exploded.

Comment (abhisheksoni27) [inline src/core/utils.js]: I am not sure I totally get what you are trying to say here. Can you elaborate the part where you say "that allows them to add their array type"?

Comment (abhisheksoni27) [inline src/core/utils.js]: I know, right? When I saw your initial comment, I was like, cool that's neat and then I ran it. :laughing:

Comment (robertleeplummerjr) [inline src/core/utils.js]: Like this:
```js
static isArray(array) {
		switch (array.constructor) {
			case Int8Array:
			case Uint8Array:
			case Uint8ClampedArray:
			case Int16Array:
			case Uint16Array:
			case Int32Array:
			case Uint32Array:
			case Float32Array:
			case Float64Array:
			case Array:
				return true;
			default:
for (let i = 0; i < arrayTypes.length; i++) {
  if (array.constructor === arrayTypes[i]) {
    return true;
  }
}
				return false;
		}
}
```

Comment (abhisheksoni27) [inline src/core/utils.js]: And we can store the different types in an Array itself, rather than using a `switch` statement. Got it,

Comment (robertleeplummerjr) [inline src/core/utils.js]: You think it is worth it?

Comment (abhisheksoni27) [inline src/core/utils.js]: It's not. It goes over the array every time, which is inefficient.
If we use `.find`, it does the same thing behind the scenes.

Comment (robertleeplummerjr) [inline src/core/utils.js]: Got it!
```js
static isArray(array) {
		switch (array.constructor) {
			case Int8Array:
			case Uint8Array:
			case Uint8ClampedArray:
			case Int16Array:
			case Uint16Array:
			case Int32Array:
			case Uint32Array:
			case Float32Array:
			case Float64Array:
			case Array:
				return true;
			default:
				return Array.isArray(array);
		}
}
```

Booya!

**Pull request merged**